pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java

```diff
@@ -578,6 +577,21 @@ public CompletableFuture<Void> closeAsync() {
         }
     }
 
+    private CompletableFuture<Void> closeManagedLedgerClientFactory() {
+        final CompletableFuture<Void> future = new CompletableFuture<>();
+        executor.execute(() -> {
+            try {
+                managedLedgerClientFactory.close();
+                future.complete(null);
+            } catch (Exception e) {
+                LOG.warn("ManagedLedgerClientFactory closing failed {}", e.getMessage());
+                future.completeExceptionally(new IllegalStateException(e));
+            }
+        });
+        return FutureUtil.addTimeoutHandling(future, Duration.ofMillis(config.getBrokerShutdownTimeoutMs()), executor,
```

Comment:
`Duration.ofMillis(Math.max(1L, getConfiguration().getBrokerShutdownTimeoutMs()))` will be better.

Reply:
fixed. thanks